src/utils/wallet.js

```diff
@@ -438,53 +647,266 @@ class Wallet {
+    }
+
+    // requestId is optional, if included the server will try to confirm requestId
+    async verifyTwoFactor(accountId, securityCode) {
+        const requestData = getRequest()
+        console.log(requestData)
+        let { requestId } = requestData
+        if (!requestId && requestId !== 0) {
+            console.log('no pending multisig requestId found, assuming account setup')
+            requestId = -1
+        }
+        // try to get a accountId for the request
+        if (!accountId) accountId = requestData.accountId || this.accountId
+        if (!accountId) {
+            console.error('no pending multisig accountId found')
+            return
+        }
+        return await this.postSignedJson('/2fa/verify', {
+            accountId,
+            securityCode,
+            requestId
         });
     }
 
-    async getRecoveryMethods() {
+    async getRecoveryMethods(account) {
+        const accountId = account ? account.accountId : this.accountId
         return {
-            accountId: this.accountId,
-            data: await this.postSignedJson('/account/recoveryMethods', { accountId: this.accountId })
+            accountId,
+            data: await this.postSignedJson('/account/recoveryMethods', { accountId }, account)
         }
     }
 
-    async setupRecoveryMessage(accountId, method, securityCode) {
-        await this.validateSecurityCode(accountId, method, securityCode);
-
-        const { seedPhrase, publicKey } = generateSeedPhrase();
+    /********************************
+    Deploys 2/3 multisig contract with keys from contract-helper and localStorage
+    @todo check account has enough near to actually do this
 
+    const tempAccount = getTempAccount()
+    // temp account was set during create account
+    if (tempAccount && tempAccount.accountId) {
+        // console.log('temp account')
+        return {
+            temp: true,
+            balance: 0,
+            accountId: tempAccount.accountId,
+        }
+    }
+    // else load account normally from api
+    ********************************/
+    async deployMultisig() {
+        const accountData = await this.loadAccount()
+        // get multisig contract to deploy
+        const contractBytes = new Uint8Array(await fetch('/multisig.wasm').then((res) => res.arrayBuffer()))
+        console.log('contractBytes', contractBytes)
+        // get account
+        const { accountId } = accountData
         const account = this.getAccount(accountId)
+        console.log(account)
+        /********************************
+        KMS
+        ********************************/
+        // 1. get localstorage keys
         const accountKeys = await account.getAccessKeys();
-        if (!accountKeys.some(it => it.public_key.endsWith(publicKey))) {
-            await account.addKey(publicKey);
+        console.log('accountKeys', accountKeys)
+        // 2. get recovery method keys
```

Comment:
let's not have comments which just repeat the method name. In most cases only comments on "why" something is done make sense, comments saying what is done or what is stored are usually better replaced by appropriate method / variable names.

Reply:
commenting steps prior to writing code and sharing steps with Patrick, removed now in twoFactor.js